src/adera/Machines/RCSController.cpp

```diff
@@ -117,27 +109,36 @@ float SysMachineRCSController::thruster_influence(Vector3 posOset, Vector3 direc
 
 void SysMachineRCSController::update_controls(ActiveScene& rScene)
 {
-    auto view = rScene.get_registry().view<MachineRCSController, ACompTransform>();
+    auto view = rScene.get_registry().view<MachineRCSController>();
 
     for (ActiveEnt ent : view)
     {
         auto& machine = view.get<MachineRCSController>(ent);
-        auto& transform = view.get<ACompTransform>(ent);
+        if (!machine.m_enable)
+        {
+            continue;
+        }
 
         using wiretype::AttitudeControl;
         using Magnum::Vector3;
 
+        // Get rigidbody ancestor and its transformation component
+        auto const* pRbAncestor =
+            SysPhysics_t::try_get_or_find_rigidbody_ancestor(rScene, ent);
+        auto const& rCompRb = rScene.reg_get<ACompRigidBody_t>(pRbAncestor->m_ancestor);
+        auto const& rCompTf = rScene.reg_get<ACompTransform>(pRbAncestor->m_ancestor);
+
         if (WireData *gimbal = machine.m_wiCommandOrient.connected_value())
         {
             AttitudeControl *attCtrl = std::get_if<AttitudeControl>(gimbal);
 
+            Matrix4 transform = pRbAncestor->m_relTransform;
+
             // TODO: RCS translation is not currently implemented, only rotation
             Vector3 commandTransl = Vector3{0.0f};
```

Comment:
Not really important, but personally i prefer to do this kind of thing as

Vector3 commandTransl{...}

over

Vector3 commandTransl = Vector3{...}

Reply:
I do too usually, but in this case I do it this way because `Vector3{0.0f}` is sort of a placeholder, when there are proper translational controls It'll be `Vector3 commandTransl = something` so I kept the left side that way as a subtle reminder. I think it makes it a bit clearer.